From fda6e93359b181a392480f3ae629a04cfe55a3df Mon Sep 17 00:00:00 2001
From: Srikanth Thokala <sthokal@xilinx.com>
Date: Wed, 18 Jun 2014 10:01:37 +0530
Subject: [PATCH 0191/1566] vdmatest: Remove vdma debug crash message of DMA
 map error

This patch fixes the following the debug crash message from
vdma test client,

[    3.634772] ------------[ cut here ]------------
[    3.639334] WARNING: CPU: 1 PID: 382 at lib/dma-debug.c:1140 check_unmap+0x72c/0x87c()
[    3.647218] xilinx-vdma 43000000.axivdma: DMA-API: device driver failed to check map error[device address=0x0000000026be3800] [size=2048 bytes] [mapped as single]
[    3.661535] Modules linked in:
[    3.664754] CPU: 1 PID: 382 Comm: dma1chan0-dma1c Not tainted 3.14.0-xilinx-13600-gd4ac892 #10
[    3.673373] [<c0015644>] (unwind_backtrace) from [<c0011980>] (show_stack+0x10/0x14)
[    3.681087] [<c0011980>] (show_stack) from [<c04b3a64>] (dump_stack+0x84/0xd8)
[    3.688283] [<c04b3a64>] (dump_stack) from [<c0022eec>] (warn_slowpath_common+0x60/0x80)
[    3.696351] [<c0022eec>] (warn_slowpath_common) from [<c0022f8c>] (warn_slowpath_fmt+0x2c/0x3c)
[    3.705032] [<c0022f8c>] (warn_slowpath_fmt) from [<c021c330>] (check_unmap+0x72c/0x87c)
[    3.713104] [<c021c330>] (check_unmap) from [<c021c57c>] (debug_dma_unmap_page+0x64/0x70)
[    3.721267] [<c021c57c>] (debug_dma_unmap_page) from [<c0247c38>] (xilinx_vdmatest_slave_func+0x94c/0xc7c)
[    3.730920] [<c0247c38>] (xilinx_vdmatest_slave_func) from [<c004279c>] (kthread+0xc8/0xdc)
[    3.739236] [<c004279c>] (kthread) from [<c000e6a8>] (ret_from_fork+0x14/0x2c)
[    3.746435] ---[ end trace 0100848993ed092e ]---
++ Starting telnet daemon[    3.751029] Mapped at:
[    3.755538]
[<c021c8a0>] debug_dma_map_page+0x48/0x120
[    3.760919]  [<c02476c4>] xilinx_vdmatest_slave_func+0x3d8/0xc7c
[    3.766912]  [<c004279c>] kthread+0xc8/0xdc
[    3.771075]  [<c000e6a8>] ret_from_fork+0x14/0x2c
[    3.775762]  [<ffffffff>] 0xffffffff
[    3.785441] dma1chan0-dma1c: terminating after 1 tests, 0 failures (status 0)

Signed-off-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/dma/xilinx/vdmatest.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/drivers/dma/xilinx/vdmatest.c b/drivers/dma/xilinx/vdmatest.c
index 54f2a90..972f2ae 100644
--- a/drivers/dma/xilinx/vdmatest.c
+++ b/drivers/dma/xilinx/vdmatest.c
@@ -300,6 +300,11 @@ static int xilinx_vdmatest_slave_func(void *data)
 							thread->dsts[i],
 							test_buf_size,
 							DMA_DEV_TO_MEM);
+
+			if (dma_mapping_error(rx_dev->dev, dma_dsts[i])) {
+				failed_tests++;
+				continue;
+			}
 			xt.dst_start = dma_dsts[i];
 			xt.dir = DMA_DEV_TO_MEM;
 			xt.numf = vsize;
@@ -315,6 +320,11 @@ static int xilinx_vdmatest_slave_func(void *data)
 
 			dma_srcs[i] = dma_map_single(tx_dev->dev, buf, len,
 							DMA_MEM_TO_DEV);
+
+			if (dma_mapping_error(tx_dev->dev, dma_srcs[i])) {
+				failed_tests++;
+				continue;
+			}
 			xt.src_start = dma_srcs[i];
 			xt.dir = DMA_MEM_TO_DEV;
 			xt.numf = vsize;
-- 
1.7.5.4

